Design.md

```diff
@@ -0,0 +1,187 @@
+---
+authors: Brian Green (bgreen007@cox.net)
+state: draft
+---
+
+## What
+Job worker service that provides an API to run arbitrary Linux processes.
+
+## Details
+
+
+### Security
+Use mTLS for gRPC communication.
+
+Create a self-signed root certificate authority (CA) to sign the certificates used by the clients and server. This CA will also be used to verify the authenticity of the certs during gRPC authentication. Using RSA 2048.
+
+Required files: Server
+| File | Description |
+| --- | --- |
+| server_cert.pem | the server's certificate (public key) |
+| server_key.pem | the server's private key |
+| root_ca.pem | the certificate of the CA that can verify the client's certificate |
+
+Required files: Client:
+| File | Description |
+| --- | --- |
+| client_cert.pem | the client's certificate (public key). |
+| client_key.pem | the client's private key. |
+| root_ca.pem | the certificate of the CA that can verify the server's certificate. |
```

Comment:
originally you had a CA for servers and another CA for clients, but now there's only one CA?

Reply:
At least personally, a single CA should be fine for this task and keeps it simpler.

Reply:
Opted for the simpler solution.